yrs/src/types/xml.rs

```diff
@@ -1,61 +1,82 @@
-use crate::block::{Block, Item, ItemContent, ItemPosition, Prelim};
-use crate::block_store::Snapshot;
+use crate::block::{Block, BlockPtr, Item, ItemContent, ItemPosition, Prelim};
+use crate::block_iter::BlockIter;
 use crate::transaction::TransactionMut;
-use crate::types::text::{Diff, TextEvent, YChange};
+use crate::types::text::TextEvent;
 use crate::types::{
-    event_change_set, event_keys, Attrs, Branch, BranchPtr, Change, ChangeSet, Delta, Entries,
-    EntryChange, Map, Observers, Path, Text, ToJson, TypePtr, Value, TYPE_REFS_XML_ELEMENT,
+    event_change_set, event_keys, Branch, BranchPtr, Change, ChangeSet, Delta, Entries,
+    EntryChange, EventHandler, MapRef, Observers, Path, ToJson, TypePtr, TYPE_REFS_XML_ELEMENT,
     TYPE_REFS_XML_FRAGMENT, TYPE_REFS_XML_TEXT,
 };
-use crate::{ReadTxn, SubscriptionId, ID};
+use crate::{Map, Observable, ReadTxn, Text, ID};
 use lib0::any::Any;
 use std::borrow::Borrow;
 use std::cell::UnsafeCell;
 use std::collections::{HashMap, HashSet};
-use std::convert::TryInto;
+use std::convert::{TryFrom, TryInto};
 use std::fmt::Write;
 use std::marker::PhantomData;
 use std::ops::{Deref, DerefMut};
 use std::rc::Rc;
-use std::sync::Arc;
+
+/// Trait shared by preliminary types that can be used as XML nodes: [XmlElementPrelim],
+/// [XmlFragmentPrelim] and [XmlTextPrelim].
+pub trait XmlPrelim: Prelim {
```

Comment:
`XmlPrelim` is specialization of `Prelim` that's limited only to XML nodes (texts, elements and fragments). This way we can limit what can be inserted into XML fragments/elements.